src/hypnos/geometry_maker.py

```diff
@@ -45,7 +45,7 @@ class GeometryMaker():
         constructed geometry
     key_route_delimiter (str): delimiter for parameter paths
     '''
-    def __init__(self) -> None:
+    def __init__(self, custom_classes=[]) -> None:
```

Comment:
This works and might well be the best solution, but it could also be possible for us to avoid needing a keyword argument by taking advantage of the fact that the python contains a list of all the subclasses which inherit from a given base class in the current scope via: `ComponentBase.__subclasses__()`

Searching that list for the correct `classname` would discover both native hypnos subclasses and user-defined subclasses. Thus the user would only need to define their subclass and make sure it's in scope (e.g. import it) and hypnos could find it.

Reply:
The following function could support this:
````python
def get_subclass_from_classname(classname) -> ComponentBase:
    """Return a ComponentBase subclass by name."""
    subclasses = ComponentBase.__subclasses__()
    for subclass in subclasses:
        if subclass.__name__ == classname:
            return subclass
    raise ValueError("No such class.")
```